rest.getinsomnia.Insomnia.json

```diff
@@ -0,0 +1,68 @@
+{
+    "app-id": "rest.getinsomnia.Insomnia",
+    "base": "io.atom.electron.BaseApp",
+    "base-version": "stable",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "Insomnia",
+    "finish-args": [
+        "--socket=x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
+        "--filesystem=home"
+    ],
+    "separate-locales": false,
+    "modules": [
+        {
+            "name": "Insomnia",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm755 apply_extra /app/bin/apply_extra",
+                "install -Dm644 rest.getinsomnia.Insomnia.png /app/share/icons/hicolor/128x128/apps/rest.getinsomnia.Insomnia.png",
+                "install -Dm644 -t /app/share/applications rest.getinsomnia.Insomnia.desktop",
+                "install -Dm644 -t /app/share/metainfo rest.getinsomnia.Insomnia.appdata.xml",
+                "install -Dm755 insomnia.sh /app/bin/Insomnia"
+            ],
+            "sources": [
+                {
+                    "type": "script",
+                    "dest-filename": "apply_extra",
+                    "commands": [
+                        "tar xf insomnia.tar.gz",
+                        "rm -f insomnia.tar.gz",
+                        "mkdir -p /app/extra/Insomnia",
+                        "mv insomnia-6.0.1/* /app/extra/Insomnia"
+                    ]
+                },
+                {          
+                    "type": "extra-data",
+                    "filename": "insomnia.tar.gz",
+                    "only-arches": ["x86_64"],
+                    "url": "https://github.com/getinsomnia/insomnia/releases/download/v6.0.1/insomnia-6.0.1.tar.gz",
```

Comment:
Given it's open source, you should really try to get it build from source with [flatpak-builder-tools](https://github.com/flatpak/flatpak-builder-tools). Also because of the same reason, it doesn't need to use `extra-data`.

Reply:
Thank you for advice. Would you share with me some apps on Flathub which uses this utility? Is this required step for open source apps like this? If no I propose to release first version based on *extra-data*.

Reply:
It is required to try and build from source. There are some known problem modules on npm that can't be built using this method - if you hit one of those, then it might be OK to go extra-data on a case by case basis.

Reply:
As it is open source, you can use `archive` type instead; additional thing to add then is another `archive` with source code to make it available via `Sources` extension.